pkg/tfbridge/tests/provider_test.go

```diff
@@ -283,6 +283,393 @@ func TestReproMinimalDiffCycle(t *testing.T) {
+		},
+	}, newTestProviderOptions{})
+
+	t.Run("diff_panic", func(t *testing.T) {
+		defer func() {
+			if r := recover(); r == nil {
+				t.Fatal("Expected diff to panic")
+			}
+		}()
+		replay.ReplaySequence(t, p, `
+	[
+		{
+			"method": "/pulumirpc.ResourceProvider/Diff",
+			"request": {
+				"urn": "urn:pulumi:dev::teststack::testprov:index:ExampleResource::exres",
+				"olds": {
+					"networkConfiguration": {
+						"__defaults": [
+						"assignPublicIp"
+						],
+						"assignPublicIp": false,
+						"securityGroups": [
+						"04da6b54-80e4-46f7-96ec-b56ff0331ba9"
+						],
+						"subnets": "[\"first\",\"second\"]"
+					}
+				},
+				"news": {
+					"networkConfiguration": {
+						"__defaults": [
+						"assignPublicIp"
+						],
+						"assignPublicIp": false,
+						"securityGroups": [
+						"04da6b54-80e4-46f7-96ec-b56ff0331ba9"
+						],
+						"subnets": "[\"first\",\"second\"]"
+					}
+				}
+			},
+			"response": {
+			}
+		}
+	]
+	`)
+	})
+
+	t.Run("diff_no_panic", func(t *testing.T) {
+		replay.ReplaySequence(t, p, fmt.Sprintf(`
+		[
+		{
+			"method": "/pulumirpc.ResourceProvider/Check",
+			"request": {
+				"urn": "urn:pulumi:dev::teststack::testprov:index:ExampleResource::exres",
+				"randomSeed": "ZCiVOcvG/CT5jx4XriguWgj2iMpQEb8P3ZLqU/AS2yg=",
+				"olds": {
+					"__defaults": []
+				},
+				"news": {
+					"networkConfiguration": {
+						"securityGroups": [
+						"04da6b54-80e4-46f7-96ec-b56ff0331ba9"
+						],
+						"subnets": "[\"first\",\"second\"]"
+					}
+				}
+			},
+			"response": {
+				"inputs": {
+					"__defaults": [],
+					"networkConfiguration": {
+						"__defaults": [
+						"assignPublicIp"
+						],
+						"assignPublicIp": false,
+						"securityGroups": [
+						"04da6b54-80e4-46f7-96ec-b56ff0331ba9"
+						],
+						"subnets": "[\"first\",\"second\"]"
+					}
+				}
+			}
+		},
+		{
+			"method": "/pulumirpc.ResourceProvider/Diff",
+			"request": {
+				"urn": "urn:pulumi:dev::teststack::testprov:index:ExampleResource::exres",
+				"olds": {
+					"networkConfiguration": {
+						"__defaults": [
+						"assignPublicIp"
+						],
+						"assignPublicIp": false,
+						"securityGroups": [
+						"04da6b54-80e4-46f7-96ec-b56ff0331ba9"
+						],
+						"subnets": "[\"first\",\"second\"]"
+					}
+				},
+				"news": {
+					"networkConfiguration": {
+						"__defaults": [
+						"assignPublicIp"
+						],
+						"assignPublicIp": false,
+						"securityGroups": [
+						"04da6b54-80e4-46f7-96ec-b56ff0331ba9"
+						],
+						"subnets": "[\"first\",\"second\"]"
+					}
+				}
+			},
+			"response": {
+			},
+			"errors": [
+			"%s"
+			]
+		}
+		]
+		`, "diffing urn:pulumi:dev::teststack::testprov:index:ExampleResource::exres: "+
+			`panicked: \"value has no attribute of that name\"`))
+	})
+
+	t.Run("update_panic", func(t *testing.T) {
+		defer func() {
+			if r := recover(); r == nil {
+				t.Fatal("Expected update to panic")
+			}
+		}()
```

Comment:
There is an `assert.Panics` function for this: https://pkg.go.dev/github.com/stretchr/testify/assert#Panics.